### PR TITLE
feat(chronicle-sink): add support for compression to Google Chronicle sink

### PR DESCRIPTION
**Pull request by ChocPanda**
Google Chronicle's ingestion API supports use of GZip compression - https://cloud.google.com/chronicle/docs/reference/ingestion-api#frequently_asked_questions

### CODE REVIEW

**Review by buraizu — APPROVED** [review]

**Comment by pront** [comment]
> [CLA assistant check](https://cla.datadoghq.com/vectordotdev/vector?pullRequest=21272) Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.datadoghq.com/vectordotdev/vector?pullRequest=21272) before we can accept your contribution.You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.datadoghq.com/check/vectordotdev/vector?pullRequest=21272) it.

Hey @ChocPanda, thank you for this contribution! 
A few comments:
* The changes look good!
* Please see the above. 
* Please take a look at https://github.com/vectordotdev/vector/blob/master/changelog.d/README.md.

File: src/sinks/gcp_chronicle/chronicle_unstructured.rs [hunk 1]
```diff
@@ -52,6 +52,35 @@ use crate::{
     tls::{TlsConfig, TlsSettings},
 };
 
+
+/// Compression configuration.
+#[configurable_component]
+#[derive(Clone, Debug, Derivative, Eq, PartialEq)]
```
Comment (bruceg) [inline src/sinks/gcp_chronicle/chronicle_unstructured.rs:58]: nit: If you add `Copy` to the list of derives, you can avoid an unnecessary `clone` below.

File: src/sinks/gcp_chronicle/chronicle_unstructured.rs [hunk 2]
```diff
@@ -52,6 +52,35 @@ use crate::{
...
+))]
+pub enum ChronicleCompression {
+    /// No compression.
+    #[derivative(Default)]
+    None,
+
+    /// [Gzip][gzip] compression.
+    ///
+    /// [gzip]: https://www.gzip.org/
+    Gzip,
```
Comment (bruceg) [inline src/sinks/gcp_chronicle/chronicle_unstructured.rs:72]: Would there be any demand to configure the compression level, given the different performance characteristics that could bring?

**Review by bruceg — APPROVED** [review]

File: src/sinks/gcp_chronicle/chronicle_unstructured.rs (hunk 2 above)
Comment (ChocPanda) [reply src/sinks/gcp_chronicle/chronicle_unstructured.rs]: I can't speak to a required demand but adding configurability doesn't seem to difficult and wouldn't add too much overhead when configuring compression so I've added it in

**Comment by pront** [comment]
@bruceg would you like to take another look before we merge this? There was a force push and multiple updates after you last reviewed.

File: src/sinks/gcp_chronicle/compression.rs [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+// Schema generation largely copied from `src/sinks/util/buffer/compression`
+impl Configurable for ChronicleCompression {
+    fn metadata() -> Metadata {
+        let mut metadata = Metadata::default();
+        metadata.set_title("Compression configuration.");
+        metadata.set_description("All compression algorithms use the default compression level unless otherwise specified.");
+        metadata.add_custom_attribute(CustomAttribute::kv("docs::enum_tagging", "external"));
+        metadata.add_custom_attribute(CustomAttribute::flag("docs::advanced"));
+        metadata
```
Comment (bruceg) [inline src/sinks/gcp_chronicle/compression.rs:64]: This looks identical to `Compression::metadata`. Does this work?
```suggestion
        Compression::metadata()
```

File: src/sinks/gcp_chronicle/compression.rs [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+    }
+
+    fn generate_schema(gen: &RefCell<SchemaGenerator>) -> Result<SchemaObject, GenerateError> {
```
Comment (bruceg) [inline src/sinks/gcp_chronicle/compression.rs:67]: Could any of this be refactored out into functions shared with `Compression`? This looks like a lot of duplication just with different lists.

@pront are there other functions that duplicate parts of this as well? Maybe that kind of clean up is out of scope here.

**Review by bruceg — COMMENTED** [review]
Clippy is requiring one change and I have suggestions for a couple of other clean ups. Otherwise this LGTM.

File: src/sinks/gcp_chronicle/compression.rs (hunk 4 above)
Comment (pront) [reply src/sinks/gcp_chronicle/compression.rs:62]: I agree there's potential for code sharing here. That being said, I am happy to accept this PR as is. We can add a comment here as a reminder to follow up and refactor.

File: changelog.d/chronicle_compression_support.enhancement.md [hunk 5]
```diff
@@ -0,0 +1,4 @@
+Add compression support to the Chronicle log sink
+Only GZip compression is documented as being supported - https://cloud.google.com/chronicle/docs/reference/ingestion-api#frequently_asked_questions
```
Comment (pront) [inline changelog.d/chronicle_compression_support.enhancement.md:2]:
```suggestion
Add Gzip compression support to the`gcp_chronicle_unstructured` sink. See official documentation [here](https://cloud.google.com/chronicle/docs/reference/ingestion-api#frequently_asked_questions).
```

**Review by pront — APPROVED** [review]

**Pull request merged**